Add documentation for agent, chatbot and MCP added to Otel-demo (#10574)
open-telemetry/opentelemetry.io: content/en/docs/demo/services/agent.md

---
title: Agent Service
linkTitle: Agent
aliases: [agentservice]
cSpell:ignore:
  langgraph langchain litellm traceloop fastmcp uvicorn gradio vcr cassettes
  ChatOpenAI
---

The Agent service provides an AI assistant for the OpenTelemetry Astronomy Shop
demo. It exposes a FastAPI HTTP endpoint that accepts user prompts, routes them
through a LangGraph ReAct agent, and uses either built-in shop tools or
MCP-provided tools to interact with the demo application.

[Agent service source](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/agent/)

## Overview

> - Runtime: Python 3.11

Reviewer: ```suggestion
- Runtime: Python 3.14
```

Whenever this one gets merged: https://github.com/open-telemetry/opentelemetry-demo/pull/3561

| `MCP_PORT`                    | `8011`                              | Port for the MCP service.                                                                                          |
| `OTEL_EXPORTER_OTLP_ENDPOINT` | `localhost:4317`                    | OTLP endpoint used by Traceloop/OpenTelemetry. In Compose this points to the OpenTelemetry Collector.              |
| `OTEL_EXPORTER_OTLP_INSECURE` | unset                               | Set to `true` in Compose for insecure local OTLP export.                                                           |
| `OTEL_RESOURCE_ATTRIBUTES`    | inherited                           | Additional OpenTelemetry resource attributes.                                                                      |
| `OTEL_SERVICE_NAME`           | `AstronomyShopAgent`                | Service name used in telemetry.                                                                                    |

Do not commit real API keys. Prefer local overrides or secret management for
`API_KEY`. Note that the VCR file is created using `LLM_MODEL` and is case
sensitive.

### Docker Compose configuration

> In `docker-compose.yml`, the service is named `agent` and is built from

Reviewer: ```suggestion
In `compose.agent.yaml`, the service is named `agent` and is built from
```

---
title: Agent Service
linkTitle: Agent

> aliases: [agentservice]

Reviewer 2: We don't need an alias here. The `xyzservice` was for redirects of old pages. This is a new page.
```suggestion
```
